src/server.py

```diff
@@ -116,33 +121,66 @@ def parse_req(request):
     return request_decon[1]
 
 
-def response_decision(full_mes):
+def response_decision(request):
     """Based on the raised error, return an appropriate http-exception
      and return URI = 'None'. Return an ok-response and URI for a
      good request. Response = byte-string, uri = unicode-string."""
     uri = u'None'
     try:
-        parse_req(full_mes)
+        parse_req(request)
     except NotImplementedError:
         response = HTTPException(u'405', u'Method Not Allowed',
-                                 u'The server supports HTTP/1.1 only.')\
-            .response_msg()
+                                 u'The server supports HTTP/1.1 only.')
+        response = response.response_msg()
     except TypeError:
         response = HTTPException(u'505', u'HTTP Version Not supported',
-                                 u'The server supports HTTP/1.1 only.')\
-            .response_msg()
+                                 u'The server supports HTTP/1.1 only.')
+        response = response.response_msg()
     except NameError:
         response = HTTPException(u'400', u'Bad Request',
-                                 u'No <host> in headers.')\
-            .response_msg()
+                                 u'No <host> in headers.')
+        response = response.response_msg()
     except:
         response = HTTPException(u'500', u'Internal Server Error',
-                                 u'Something went wrong.')\
-            .response_msg()
+                                 u'Something went wrong.')
+        response = response.response_msg()
+    else:
+        uri = parse_req(request)
+        response = generate_response(uri)
+    return response
+
+
+def resolve_uri(uri):
+    """Return body and mimetype for a given uri or False if
+     source not found."""
+    path = ABS_PATH + uri
+    if os.path.isfile(path):
+        files = io.open(path, 'rb')
+        mime = guess_type(path)
+        read_file = files.read()
+        files.close()
+        return (read_file, mime)
+    elif os.path.isdir(path):
+        return (str(os.listdir(path)), None)
     else:
-        response = response_ok()
-        uri = parse_req(full_mes)
-    return [response, uri]
+        return False
+
+
+def generate_response(uri):
+    """Generate a response based on the return of resolve_uri(uri):
+     a. - ok-message with body and mime if file was found,
+     b. - 404 Not Found error msg if file was not found."""
+    if resolve_uri(uri):
+        body, mime = resolve_uri(uri)
+        response = response_ok(body, mime)
+    else:
+        response = HTTPException(
+            u'404',
+            u'Resource Not Found',
+            u"HTTP Error 404: I can't find what you are looking for.")
+        response = response.response_msg()
+    return response
```

Comment:
This function will return a good response, or an error response.  Harder to test than it might be.